python/main.py

```diff
@@ -9,24 +9,30 @@
 
 with open(temp_file, 'r') as f:
     print(f)
+    
 
 os.chmod("/tmp/foo.txt", stat.S_IXGRP)
 tar_file = '/file.tax*'
 os.system(tar_file)
 
+
 KEY_SIZE = 1024
+
```

Comment:
SAST

<details><summary><img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsSeverityIcons/medium_new.svg" alt="MEDIUM" width="24px" align="center">&nbsp;&nbsp;<b style="color:black" >External control of system or configuration setting</b><br>&nbsp;&nbsp;&nbsp;&nbsp;Source File: <a href="https://MOCK_DOMAIN/bridgecrew/platform/blob/my-awesome-branch/ExternalControlOfSystemOrConfigurationSetting1-taint3.java#L13-L13">ExternalControlOfSystemOrConfigurationSetting1-taint3.java</a>, Sink File: <a href="https://MOCK_DOMAIN/bridgecrew/platform/blob/my-awesome-branch/ExternalControlOfSystemOrConfigurationSetting1-taint3.java#L14-L14">ExternalControlOfSystemOrConfigurationSetting1-taint3.java</a> | Checkov ID: <a href="https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-code-security-policy-reference/sast-policies/java-policies/sast-policy-108">CKV3_SAST_108</a></summary>
<br><details><summary><b>Data Flow</b></summary>

&nbsp;&nbsp;&nbsp;&nbsp; ExternalControlOfSystemOrConfigurationSetting1-taint3.java
&nbsp;&nbsp;&nbsp;&nbsp;`13 | String tainted = req.getParameter("input");`
&nbsp;&nbsp;&nbsp;&nbsp; ExternalControlOfSystemOrConfigurationSetting1-taint3.java
&nbsp;&nbsp;&nbsp;&nbsp;`14 | c.setCatalog(tainted);`

</details>
<br><details><summary><b>How To Fix</b></summary>

String userInput = request.getParameter("catalogName");
if(isValidCatalogName(userInput)) {
    connection.setCatalog(userInput);
}

</details>
<br><details><summary><h4>Description</h4>

</summary>


This SAST policy detects instances where user input is used in the `java.sql.Connection`'s `setCatalog` call. This can be a potential issue as it may allow attackers to change the database during a connection, leading to disruptions or unexpected behaviors.

Vulnerable code example:

```
java
String userInput = request.getParameter("catalogName");
connection.setCatalog(userInput);
```

In the above code, the `setCatalog` method uses user input `catalogName` directly without any validation or sanitization. This allows an attacker to manipulate the `catalogName` parameter and potentially change the database being accessed.


</details>


</details>